### PR TITLE
Sign up updates

### PR DESCRIPTION
**Pull request by AllBill127**
Redo SignUpPage with delegates
Fix minor issues

### CODE REVIEW

File: CityPuzzle/Login And SingUp/SignUpPage.xaml.cs [hunk 1]
```diff
@@ -21,70 +18,47 @@ public SignUpPage()
...
-                        if (!Valid_Username.IsMatch(field.Text)) throw new BadInputdException("nuo 6 iki 12 ilgio", field.Placeholder, field);
-                        else if (!User.CheckUser(field.Text)) throw new BadInputdException("Naudotojas su tokiu prisijungimo vardu užimtas.Pakeiskite prisijungimo vardą ir bandykite dar kartą.", field);
+                        if (!validateUsername(field.Text)) 
+                            throw new BadInputdException("nuo 6 iki 12 ilgio", field.Placeholder, field);
+                        else if (!User.CheckUser(field.Text)) 
+                            throw new BadInputdException("Naudotojas su tokiu prisijungimo vardu užimtas.Pakeiskite prisijungimo vardą ir bandykite dar kartą.", field);
+                        break;
+                    case "Slaptazodis":
+                        if (!validatePassword(field.Text)) 
+                            throw new BadInputdException("Password must be atleast 8 to 15 characters and contains atleast one Upper case and numbers.", field);
```
Comment (crmsnwl) [inline CityPuzzle/Login And SingUp/SignUpPage.xaml.cs:46]: Error messages need editing; should be in one language.

File: CityPuzzle/Login And SingUp/SignUpPage.xaml.cs [hunk 2]
```diff
@@ -132,21 +106,18 @@ void Registration_Click(object sender, EventArgs e)
...
-            List<Lazy<Puzzle>> zero = new List<Lazy<Puzzle>>();
             User user = new User()
             {
                 Name = name,
                 LastName = lastName,
                 UserName = userName,
-                Pass = password,
                 Email = email,
-                MaxQuestDistance = maxDist,
-                QuestsCompleted = zero
```
Comment (crmsnwl) [inline CityPuzzle/Login And SingUp/SignUpPage.xaml.cs:149]: Why is QuestsCompleted deleted? Are you sure it's not needed?

File: CityPuzzle/Classes/Sql.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ namespace CityPuzzle.Classes
 {
     class Sql
     {
- m        public static string ConnStr = "Server=tcp:citypuzzle.database.windows.net,1433;Initial Catalog=CityPuzzle;Persist Security Info=False;User ID=citypuzzle;Password=User123*;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;";
+        public static string ConnStr = "Server=tcp:citypuzzle.database.windows.net,1433;Initial Catalog=CityPuzzle;Persist Security Info=False;User ID=citypuzzle;Password=User123*;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;";
```
Comment (crmsnwl) [inline CityPuzzle/Classes/Sql.cs:15]: It's nothing serious, but I think there should be spaces after semicolons and around "=" to maintain a consistent code style.

File: CityPuzzle/Login And SingUp/SignUpPage.xaml.cs (hunk 2 above)
Comment (AllBill127) [reply CityPuzzle/Login And SingUp/SignUpPage.xaml.cs:149]: User class instantiates QuestsCompleted list by default on creation

File: CityPuzzle/Login And SingUp/SignUpPage.xaml.cs [hunk 4]
```diff
@@ -21,70 +18,47 @@ public SignUpPage()
...
         public static void Validation(List<Entry> fields)
         {
+            Validator validateUsername = new Validator(RegexDelegate.validUsername);
+            Validator validatePassword = new Validator(RegexDelegate.validPassword);
+            Validator validateEmail = new Validator(RegexDelegate.validEmail);
+
             foreach (Entry field in fields)
             {
-                if (String.IsNullOrWhiteSpace(field.Text)) throw new EmptyInputdException(field.Placeholder, field);
+                if (String.IsNullOrWhiteSpace(field.Text))
```
Comment (crmsnwl) [inline CityPuzzle/Login And SingUp/SignUpPage.xaml.cs:32]: string should not start with a capital S.

File: CityPuzzle/Classes/Sql.cs [hunk 5]
```diff
@@ -12,7 +12,7 @@ namespace CityPuzzle.Classes
 {
     class Sql
```
Comment (crmsnwl) [inline CityPuzzle/Classes/Sql.cs:13]: Should probably be public

File: CityPuzzle/Login And SingUp/SignUpPage.xaml.cs (hunk 1 above)
Comment (AllBill127) [reply CityPuzzle/Login And SingUp/SignUpPage.xaml.cs]: Updated all messages to be in Lithuanian

File: CityPuzzle/Classes/Sql.cs (hunk 3 above)
Comment (AllBill127) [reply CityPuzzle/Classes/Sql.cs]: Fixed and added line breaks for better readability

Comment (crmsnwl) [reply CityPuzzle/Classes/Sql.cs]: Way more readable now.

File: CityPuzzle/Login And SingUp/SignUpPage.xaml.cs (hunk 2 above)
Comment (crmsnwl) [reply CityPuzzle/Login And SingUp/SignUpPage.xaml.cs:149]: Checked it out, everything is fine.

**Pull request merged**